### ISSUE TITLE
Switch to Non-blocking Popup window

### ISSUE THREAD

**Issue by tobiasKaminsky** [owncloud/android#828]
Switch to non-blocking popup window.
That is, having a bit of space to the sides to signify it’s only a popup?

This should be done for:
- [ ] UploadFilesActivity (upload files from within App)
- [ ] InstantUpload Path Chooser
- [ ] Uploader (upload files from other apps)

**Comment by strugee** [comment]
> UploadFilesActivity (upload files from within App)
> InstantUpload Path Chooser

Maybe these shouldn't be non-blocking. When you upload files through the Share button, you're temporarily switching context into ownCloud, so it makes sense to show some of the old context, especially since you're going to be switching back to that context as soon as you're done uploading the file.

But if you're actually in the ownCloud app, you're not switching contexts. The context is the same, you're managing your ownCloud. So it doesn't really make sense to have an indication (a smaller popup with border) that your "new" context is temporary, because it isn't new and it isn't temporary.

**Comment by tobiasKaminsky** [comment]
@jancborchardt This was your idea ;) ;)

**Comment by jancborchardt** [comment]
Choosing a path is always temporary, be it for uploading or for choosing an instant upload path. Hence the path chooser should always be a popup window which makes it clear it’s temporary.

Just to make sure we’re on the same page, this is what I mean:
![](https://i.stack.imgur.com/xot6b.png)

**Comment by tobiasKaminsky** [comment]
Yeah. And this seems to be the only thing that is possible.
A dialog can not have an action bar (at least not in Sherlock).
So we have to add a "up" button as the first item

**Comment by strugee** [comment]
> Choosing a path is always temporary, be it for uploading or for choosing an instant upload path.

Of course. But choosing a path is not always in a temporary context. If you're in the settings screen, you are configuring settings. If you're choosing what the setting for the instant upload path is, you're still configuring settings.

**Comment by jancborchardt** [comment]
Yeah, but the popup to choose that path should not look like a completely new screen. It should overlay the settings in a manner that you know where you are.

It’s an important usability concept to give people a sense of location, where they are in the app. These non-fully popups work well for that.

**Comment by davivel** [comment]
I don't like much this approach. Screen surface in smartphones is scarce, so leaving unused margins is an important lose. Specially if we have to add extra entries in the list, such as an 'Up' button to browse to the parent. And more specially if we are rendering a list of items.

In terms of style, it doesn't fit well Material Design, that we are embracing.

@jancborchardt , I guess your smartphone holds Android < 4.4, right? From Android 4.4 seems there is no way to get that screen since Dropbox just offers the standard document browser from the device, that is full screen. So does GDrive, and probably any other modern app. Not that I want to remove completely our file-browsing view, but I think it's a hint of which browsing patterns are staying in recent Android versions, and which ones are leaving.

Besides, moving through full screen activities is the most basic navigation pattern in Android (even in iOS too, AFAIK), and users are perfectly accustomed to it.

I'd search for other visual tools to provide context. I think it's important, and improvable in our app, but I don't think this is the way. Honestly, I'd wish have released the 'Share' activity as a full screen view.

**Comment by jesmrec** [comment]
Pop up windows not used.
